src/WebJobs.Script/Description/CSharp/FunctionMetadataResolver.cs

```diff
@@ -37,9 +37,15 @@ public sealed class FunctionMetadataResolver : MetadataReferenceResolver
                 "System.Xml",
                 "System.Net.Http",
                 typeof(object).Assembly.Location,
-                typeof(TraceWriter).Assembly.Location,
-                typeof(TimerInfo).Assembly.Location,
-                typeof(System.Web.Http.ApiController).Assembly.Location
+                typeof(IAsyncCollector<>).Assembly.Location, /*Microsoft.Azure.WebJobs*/
+                typeof(JobHost).Assembly.Location, /*Microsoft.Azure.WebJobs.Host*/
+                typeof(CoreJobHostConfigurationExtensions).Assembly.Location, /*Microsoft.Azure.WebJobs.Extensions*/
+                typeof(System.Web.Http.ApiController).Assembly.Location /*System.Web.Http*/
+            };
+
+        private static readonly Assembly[] SharedAssemblies =
+            {
+                typeof(Newtonsoft.Json.JsonConvert).Assembly /*Newtonsoft.Json*/
             };
```

Comment:
Default assemblies are automatically added, without the need to explicitly have a "#r" reference. Shared assemblies are the ones we make available to you _if_ you ask for it (e.g. #r "Newtonsoft.Json") but are not automatically injected. This minimizes the number of compile time references that may be unnecessary and make it simpler for you to reference a different version of a given assembly if you want to go that route.